src/vs/platform/secrets/electron-sandbox/secretStorageService.ts

```diff
@@ -0,0 +1,82 @@
+import { ILogService } from 'vs/platform/log/common/log';
+import { INativeHostService } from 'vs/platform/native/common/native';
+import { INotificationService, IPromptChoice } from 'vs/platform/notification/common/notification';
+import { IOpenerService } from 'vs/platform/opener/common/opener';
+import { BaseSecretStorageService } from 'vs/platform/secrets/common/secrets';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+
+export class NativeSecretStorageService extends BaseSecretStorageService {
+
+	constructor(
+		@INotificationService private readonly _notificationService: INotificationService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IStorageService storageService: IStorageService,
+		@IEncryptionService encryptionService: IEncryptionService,
+		@INativeHostService private readonly _nativeHostService: INativeHostService,
+		@ILogService logService: ILogService
+	) {
+		super(storageService, encryptionService, logService);
+	}
+
+	override set(key: string, value: string): Promise<void> {
+		this._sequencer.queue(key, async () => {
+			await this.initialized;
+
+			if (this.type !== 'persisted') {
+				this._logService.trace('[NativeSecretStorageService] Notifying user that secrets are not being stored on disk.');
+				await this.notifyOfNoEncryptionOnce();
+			}
+
+		});
+
+		return this._sequencer.queue(key, () => super.set(key, value));
+	}
+
+	private notifyOfNoEncryptionOnce = once(() => this.notifyOfNoEncryption());
+	private async notifyOfNoEncryption(): Promise<void> {
+		const buttons: IPromptChoice[] = [];
+		const troubleshootingButton: IPromptChoice = {
+			label: localize('troubleshootingButton', "Open troubleshooting guide"),
+			run: () => this._instantiationService.invokeFunction(accessor => accessor.get(IOpenerService).open('https://go.microsoft.com/fwlink/?linkid=2239490')),
+			keepOpen: true
+		};
+		buttons.push(troubleshootingButton);
+
+		let errorMessage = localize('encryptionNotAvailableJustTroubleshootingGuide', "Secrets are not being stored on disk because encryption is not available in this environment.");
+
+		if (!isLinux) {
+			this._notificationService.prompt(Severity.Error, errorMessage, buttons);
+			return;
+		}
+
+		const provider = await this._encryptionService.getKeyStorageProvider();
+		if (isGnome(provider)) {
+			errorMessage = localize('isGnome', "You're running in a GNOME environment but encryption is not available. Ensure you have gnome-keyring or another libsecret compatible implementation installed and running.");
+		} else if (isKwallet(provider)) {
+			errorMessage = localize('isKwallet', "You're running in a KDE environment but encryption is not available. Ensure you have kwallet running.");
+		} else if (provider === KnownStorageProvider.basicText) {
+			errorMessage += ' ' + localize('usePlainTextExtraSentence', "Open the troubleshooting guide or you can use plain text obfuscation instead.");
```

Comment:
An average user might not know what `plain text obfuscation` would be ? Maybe a better wording would be

```
An OS keyring couldn't be identified for storing the encryption related data in your current desktop
environment, please open the troubleshooting guide to address this or you can use a in-memory based weaker encryption.
```